# DFSurv
The code about DFSurv: A Diffusion-Transformer Synergistic Network for Robust and Interpretable Multi-Omics Cancer Prognosis Analysis

**Python Version:** 3.8  

**Library Dependencies:**
captum                            0.8.0
Tensorflow                        2.16.1
Torch: 2.5.1
Torch-Geometric: 2.6.1
torch-scatter                     2.1.2
torch-sparse                      0.6.18
tqdm                              4.65.0  

**Data preparation**
In this study, we utilized cancer datasets from the TCGA portal (https://tcga-data.nci.nih.gov/tcga/). All these datasets were downloaded by using the R package “TCGA-assembler”(v1.0.3, (Wei, et al., 2018)), which contains four types of multi-omics data: mRNA, miRNA, and DNA methylation. Here, “mRNA” was RNA sequencing data generated by UNC Illumina HiSeq_RNASeq V2; Level 3, “miRNA” was miRNA sequencing data obtained by BCGSC Illumina HiSeq miRNASeq, DNA methylation data was generated by USC HumanMethylation450.

# Example
For easy to use, here we give cancer multi-omics data : test_mrna as mrna data, test_meth as methylation data, test_mi as mirna data. Run train_pipeline.py for 5-fold CV evaluation.

## Python Files Overview:
train_pipeline.py: This is the primary framework file, run for cancer survival outcome prediction. The corresponding result can be found in the correspongding folder named "output_XXX".

<img width="990" height="484" alt="image" src="https://github.com/user-attachments/assets/84e0d751-7510-4f78-8ca4-32ac182084bf" />

# Contact
This method is still on progress, any questions can be sent to oajumping@163.com
